### PR TITLE
Stop looking for DNS records by data only

### PR DESCRIPTION
**Pull request by mpolden**
Merge with internal PR.

@jonmv

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java [hunk 1]
```diff
@@ -64,32 +55,18 @@ public Optional<RecordData> data() {
...
-            name.ifPresent(n -> records.addAll(nameService.findRecords(type, n)));
-            data.ifPresent(d -> records.addAll(nameService.findRecords(type, d)));
-        }
-        nameService.removeRecords(records);
+        // Deletions require all records fields to match exactly, data may be incomplete even if present. To ensure
+        // completeness we search for the record(s) first
+        List<Record> completeRecords = nameService.findRecords(type, name).stream()
+                                                  .filter(record -> data.isEmpty() || matchingFqdnIn(data.get(), record))
+                                                  .toList();
+        nameService.removeRecords(completeRecords);
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java:63]: So we just drop deletion of unmatched records? Should we log a warning when this happens?

**Review by jonmv — APPROVED** [review]
One comment, looks good.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java (hunk 1 above)
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java:63]: The logic should be the same as before this change. Not sure what you mean by unmatched here? If you specify both name+data you specifically want to only delete records matching both fields.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java:63]: Aha, that sounds right.

**Pull request merged**